src/calculation/DetailedResultsBuilder.ts

```diff
@@ -178,44 +210,49 @@ export function handlePopulationValues(
-    if (hasResult(PopulationType.MSRPOPL, populationResults)) {
-      const popResult = populationResults.find(result => result.populationType === PopulationType.OBSERV);
-      if (popResult) {
-        popResult.result = false;
-        popResult.observations = null;
-      }
-    }
+    // Cannot be in the denominator exception if they are excluded from the denominator
+  } else if (getResult(PopulationType.DENEX, populationResults)) {
+    setResult(PopulationType.DENEXCEP, false, populationResults);
+  }
 
-    // If the numer is influenced by the DENOM, false it out and its observations since we are not in the DENOM for this branch of logic
-    if (!MeasureBundleHelpers.isRatioMeasure(group, measureScoringCode)) {
-      setResult(PopulationType.NUMER, false, populationResults);
-      setResult(PopulationType.NUMEX, false, populationResults);
-      // If there are not multiple IPPs, then NUMER depends on DENOM. We're not in the DENOM, so let's null out NUMER observations
-      DetailedResultsHelpers.nullCriteriaRefMeasureObs(group, populationResults, PopulationType.NUMER);
-    }
+  // Cannot be in the NUMEX if not in the NUMER
+  if (!getResult(PopulationType.NUMER, populationResults)) {
+    setResult(PopulationType.NUMEX, false, populationResults);
+    // Not in NUMER, so no need for NUMER observations
+    DetailedResultsHelpers.nullCriteriaRefMeasureObs(group, populationResults, PopulationType.NUMER);
+  }
 
-    setResult(PopulationType.MSRPOPLEX, false, populationResults);
+  return populationResultsHandled;
+}
 
-    // Cannot be in the numerator if they are excluded from the denominator
-  } else if (getResult(PopulationType.DENEX, populationResults)) {
-    if (!MeasureBundleHelpers.isRatioMeasure(group, measureScoringCode)) {
-      setResult(PopulationType.NUMER, false, populationResults);
-      setResult(PopulationType.NUMEX, false, populationResults);
-      // Since we can't be in the numerator, null out numerator observations
-      DetailedResultsHelpers.nullCriteriaRefMeasureObs(group, populationResults, PopulationType.NUMER);
+export function handleCVPopulationValues(populationResults: PopulationResult[]): PopulationResult[] {
+  const populationResultsHandled = populationResults;
+  // Cannot be in all populations if not in IPP.
+  if (!getResult(PopulationType.IPP, populationResults)) {
+    populationResults.forEach(result => {
+      if (result.populationType === PopulationType.OBSERV) {
+        result.observations = null;
+      }
+      result.result = false;
+    });
+    // Short-circuit return since no more processing needs to be done if IPP is false with only one IPP
+    return populationResultsHandled;
+  }
+  // Cannot be in most populations if not in DENOM or MSRPOPL
+  if (hasResult(PopulationType.MSRPOPL, populationResults) && !getResult(PopulationType.MSRPOPL, populationResults)) {
+    // If there is a MSRPOPL, all observations point to it, so null them out
```

Comment:
I am a little confused about this logic, could you maybe walk me through it?

To my understanding, this is looking if the MSRPOPL has a result (result is not null) AND that that result is false ?

Reply:
Yes- I believe so. Happy to look at it together.